Remove docker images on make clean_all

#21 nominally added this feature but when I ran it I found the following error message:

```ellen@ellen-desktop:~/dev/headway$ make clean_all
rm -rf /home/ellen/dev/headway/data/*.mbtiles
rm -rf /home/ellen/dev/headway/data/*.nominatim.sql
rm -rf /home/ellen/dev/headway/data/*.osm.pbf
rm -rf /home/ellen/dev/headway/data/*.photon
rm -rf /home/ellen/dev/headway/data/bbox.txt
rm -rf /home/ellen/dev/headway/data/sources
rm -rf /home/ellen/dev/headway/data/nominatim_flatnode
rm -rf /home/ellen/dev/headway/data/nominatim_pg
docker images | grep ^headway_build_ | tr -s ' ' | cut -d' ' -f3 | xargs docker rmi
Error response from daemon: conflict: unable to delete f7354fa160e3 (must be forced) - image is referenced in multiple repositories
Error response from daemon: conflict: unable to delete 32a0db2e6aa4 (must be forced) - image is referenced in multiple repositories
Error response from daemon: conflict: unable to delete 7687b7da0d06 (must be forced) - image is referenced in multiple repositories
Error response from daemon: conflict: unable to delete 7f5243e4dfa0 (must be forced) - image is referenced in multiple repositories
make: *** [Makefile:150: clean_all] Error 123
```

--- a/Makefile
+++ b/Makefile
@@ -126,6 +126,11 @@ tag_images: nginx_image photon_image nominatim_image otp_image valhalla_image
 $(filter %,$(CITIES)): %: ${DATA_DIR}/%.osm.pbf ${DATA_DIR}/%.nominatim.sql ${DATA_DIR}/%.nominatim_tokenizer.tgz ${DATA_DIR}/%.photon.tgz ${DATA_DIR}/%.mbtiles ${DATA_DIR}/%.graph.obj ${DATA_DIR}/%.gtfs.tar ${DATA_DIR}/%.valhalla.tar ${DATA_DIR}/%.bbox tag_images
 	@echo "Built $@"
 
+%.up: %
+	docker-compose kill || echo "Containers not up"
+	docker-compose down || echo "Containers dont exist"
+	docker-compose up -d
+
 # Clean only generated data.
 clean:
 	rm -rf ${DATA_DIR}/*.mbtiles
@@ -136,11 +141,6 @@ clean:
 	rm -rf ${DATA_DIR}/*.graph.obj
 	rm -rf ${DATA_DIR}/bbox.txt
 	rm -rf ${DATA_DIR}/bbox.txt.bak
-
-%.up: %
-	docker-compose kill || echo "Containers not up"
-	docker-compose down || echo "Containers dont exist"
-	docker-compose up -d
 
 # Clean even the data we have to download from external sources.
 clean_all: clean

--- a/Makefile
+++ b/Makefile
@@ -147,3 +147,4 @@ clean_all: clean
 	rm -rf ${DATA_DIR}/*.osm.pbf
 	rm -rf ${DATA_DIR}/*.gtfs.tar
 	rm -rf ${DATA_DIR}/sources
+	docker images -qf "reference=headway_build_*" --format='{{.Repository}}:{{.Tag}}' | xargs docker rmi

